### ISSUE TITLE
How to get expires_in?

### ISSUE THREAD

**Issue by RazerMoon** [nextauthjs/adapters#48]
Logging the account object in a callback:

```js
async signIn(user, account, profile) {
  console.dir(account)
  return true
}
```

I get this:

```js
{
  provider: 'discord',
  type: 'oauth',
  /* ... */
  accessTokenExpires: null,
  /* ... */
  expires_in: 604800,
  /* ... */
}
```

`accessTokenExpires` is null, so I want to use `expires_in` instead, but linkAccount doesn't provide that.

https://github.com/nextauthjs/adapters/blob/4dcadf99414485ce55e32a8df79deb65854cbb84/packages/example/index.js#L40

I know I could probably hard code the value but I'd rather not if I can help it.

Any suggestions? Am I doing this wrong? Thanks in advance.

**Comment by balazsorban44** [comment]
You are right. `accessTokenExpires` has been set to `null` for historical reasons. The `linkAccount` signature could really use a named parameter syntax, and we should just send all the tokens we get from the provider... 🤦

Unfortunately, this is not something I have the time to implement, but PRs are welcome!

@iaincollins or @ndom91 maybe could guide you where this change should be made.

**Comment by sastraxi** [comment]
How would one even implement the example token rotation code then -- is it just impossible if you're using a database adapter? https://next-auth.js.org/tutorials/refresh-token-rotation

**Comment by balazsorban44** [comment]
currently, I think so. the account has been sent to the adapter very selectively by the original creator for some reason. We intend to change this in the next release.

**Comment by sastraxi** [comment]
> you could use the `signIn` callback or event to write the expires_in value to your database manually

This is exactly what I've done and it works well enough for now. Thanks for confirming, and good luck with the next release -- it already looks like it's going to address a lot of my pain points with v3.
